### PR TITLE
Logging and cred provider 

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/docker/MavenDockerCrendetialProvider.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.settings.Server;
+import org.apache.maven.settings.Settings;
+
+public class MavenDockerCrendetialProvider implements IDockerCredentialProvider {
+    private Server server;
+
+    public MavenDockerCrendetialProvider(Settings settings, String serverId) throws ServerNotFoundException {
+        if (StringUtils.isNotBlank(serverId)) {
```
Comment (Flanker32) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/docker/MavenDockerCrendetialProvider.java]: should also check `settings!=null`

File: azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public void info(String content, Throwable error) {
+        print(System.out, content, error);
+    }
+
+    @Override
+    public void warn(String content) {
+
```
Comment (Flanker32) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java]: Didn't implement yet?

File: azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public void error(Throwable error) {
+
+    }
+
+    private void print(PrintStream output, String content, Throwable error) {
+        if (content != null && content.length() > 0) {
```
Comment (Flanker32) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java]: `content != null && content.length() > 0` -> `StringUtils.isNotEmpty` ?

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/docker/MavenDockerCrendetialProvider.java (hunk 1 above)
Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/docker/MavenDockerCrendetialProvider.java]: good catch

File: azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java (hunk 2 above)
Comment (andxu) [reply azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java]: Done

File: azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java (hunk 3 above)
Comment (andxu) [reply azure-tools-common/src/main/java/com/microsoft/azure/common/logging/DefaultLogger.java]: Done

**Pull request merged**